From 3632aaf8a7ec5bc15fc3d9b296407059ce142839 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Val=C3=A9rio=20Val=C3=A9rio?= <valerio.valerio@jolla.com>
Date: Mon, 23 Feb 2015 13:08:24 +0200
Subject: [PATCH 01/21] Stop _incomingDataTimer when imapprotocol object is
 destroyed.

A crash occurs in case the timer is active and the object is destroyed.
---
 src/plugins/messageservices/imap/imapprotocol.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/plugins/messageservices/imap/imapprotocol.cpp b/src/plugins/messageservices/imap/imapprotocol.cpp
index 630b16ed..26b7837a 100644
--- a/src/plugins/messageservices/imap/imapprotocol.cpp
+++ b/src/plugins/messageservices/imap/imapprotocol.cpp
@@ -3012,6 +3012,9 @@ ImapProtocol::ImapProtocol()
 
 ImapProtocol::~ImapProtocol()
 {
+    // FIXME: does this have any effect or can it be removed? shouldn't be able to trigger
+    // before it's automatically stopped at the end of this function
+    _incomingDataTimer.stop();
     delete _transport;
     delete _fsm;
 }
-- 
2.17.1

